### ISSUE TITLE
Mini-gallery plugin values can't be updated without full http request.

### ISSUE THREAD

**Issue by david-nowinski** [FLEXIcontent/flexicontent-cck#463]
Hello,

The mini-gallery plugin is using the global $_FILES in its onBeforeSaveField event. The problem is that if you try to create or update the item through custom scripting, it'll simply crash because the $_FILES is not filled. (I'm using the [method in the FAQ](http://www.flexicontent.org/documentation/faq/70-developer-api-field-plugins/580-how-to-create-a-flexicontent-item-by-calling-the-flexicontent-item-model-in-custom-code.html))

To fix, I had to modify flexicontent's minigallery plugin file. I don't know how to pull request, so I'm just indicating the fix below for Flexicontent version 3.0.13, in the file plugin/flexicontent_fields/minigallery/minigallery.php at line 776 and 777, you just need to replace by this :
```php
$err_code = !empty($_FILES)?$_FILES["custom"]["error"][$field->name][$n]['file-data']:UPLOAD_ERR_NO_FILE;
$new_file = ($file_id == 0);
```
It won't affect the plugin behavior in any way.

It will be now possible to use a file defined in the database :
```php
$custom['gallery'][0]['file-id'] = 1;
$custom['gallery'][1]['file-id'] = 2;
$custom['gallery'][2]['file-id'] = 3;
```
And the field values will be filled with the specified file without errors.

What do you think of this ?

**Comment by ggppdk** [comment]
It should be possible to allow field to save without a form submit so this should be fixed for both 
- minigallery
- file
fields

**Comment by ggppdk** [comment]
Fixed with 3385aea
